### ISSUE TITLE
Images from formatters inside cmd

### ISSUE THREAD

**Issue by jcubic** [jcubic/jquery.terminal#519]
### Expected behavior

Images should be rendered inside cmd

### Actual behavior

Bracket is rendered after the image

### Steps to reproduce

```javascript
$.terminal.defaults.formatters = [[/:\)/g, '[[@;;;;https://unpkg.com/emoji-datasource-twitter@4.1.0/img/twitter/64/1f603.png]]']]
term.insert(':)');
```

I doubt anyone would like to render images in cmd but this should work since formatter can return any formatting including images.

**Comment by jcubic** [comment]
Also cmd don't render attributes for some reason:

```javascript
$.terminal.defaults.formatters = [[/:\)/g, '[[@;;;;https://unpkg.com/emoji-datasource-twitter@4.1.0/img/twitter/64/1f603.png;{"style":"width:32px"}]]']]
term.settings().allowedAttributes.push('style')
term.insert(':)');
```

with echo it works fine.

**Comment by jcubic** [comment]
Bracket fixed by #550 but one issue is that you can't put cursor after the command. it always at last character.
